Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -19,36 +20,103 @@ public GenericOpenVRController(TrackingState trackingState, Handedness controlle
         /// <summary>
         /// The current source state reading for this OpenVR Controller.
         /// </summary>
-        public XRNodeState LastStateReading;
+        public XRNodeState LastStateReading { get; protected set; }
 
+        private Vector2 dualAxisPosition = Vector2.zero;
         private Vector3 currentControllerPosition = Vector3.zero;
         private Quaternion currentControllerRotation = Quaternion.identity;
-        private MixedRealityPose currentControllerData = MixedRealityPose.ZeroIdentity;
+        private MixedRealityPose lastControllerPose = MixedRealityPose.ZeroIdentity;
+        private MixedRealityPose currentControllerPose = MixedRealityPose.ZeroIdentity;
 
-        //TODO - Do we need a pointer Pose simulator?
-        //private Vector3 currentPointerPosition = Vector3.zero;
-        //private Quaternion currentPointerRotation = Quaternion.identity;
-        //private MixedRealityPose currentPointerData = MixedRealityPose.ZeroIdentity;
-
-
-        //TODO - Do we need a Grip Pose simulator?
-        //private Vector3 currentGripPosition = Vector3.zero;
-        //private Quaternion currentGripRotation = Quaternion.identity;
-        //private MixedRealityPose currentGripData = MixedRealityPose.ZeroIdentity;
-
-        #region Base override configuration
-
-        /// <summary>
-        /// Mapping method to expose the Unity Input Manager mapping configuration
-        /// </summary>
-        public override InputManagerAxis[] ControllerAxisMappings => ControllerInputAxisMappingLibrary.GetInputManagerAxes(GetType().FullName);
-
-        /// <summary>
-        /// Mapping method to expose this controllers Unity Input Manager mapping array
-        /// </summary>
-        public virtual string[] VRInputMappings => ControllerInputAxisMappingLibrary.GetInputManagerMappings(GetType().FullName);
-
-        #endregion Base override configuration
+        public static readonly MixedRealityInteractionMapping[] DefaultLeftHandedInteractions =
```

Comment:
Are these expected to be an exhaustive list? It seems like Menu is missing.

Reply:
Menu is just another button.

Reply:
It's also a defined `DeviceInputType` not represented here.

![image](https://user-images.githubusercontent.com/3580640/43105129-aa8d8d48-8e88-11e8-97f2-1b3050d7e4d1.png)

Reply:
These only relate to what the OpenVR standard that Unity implements.

We may update this when/if we change to getting input direct from the driver.